starlite/contrib/htmx/response.py

```diff
@@ -0,0 +1,157 @@
+        and pass redirect url
+        """
+        super().__init__(
+            content=None,
+            status_code=HTTP_200_OK,
+            headers={"HX-Redirect": quote(url, safe="/#%[]=:;$&()+,!?*@'~"), "Location": ""},
+        )
+        del self.headers["Location"]
+
+
+class ClientRefresh(Response):
+    """Class to support HTMX client page refresh"""
+
+    def __init__(self) -> None:
+        """Set Status code to 200 and set headers."""
+        super().__init__(content=None, status_code=HTTP_200_OK, headers={"HX-Refresh": "true"})
+
+
+class PushUrl(Response):
+    """Class to push new url into the history stack"""
+
+    def __init__(self, content: T, url: str = "", **kwargs: Any) -> None:
+        """Initialize"""
+        push = "false" if url == "" else url
+        super().__init__(content=content, status_code=HTTP_200_OK, headers={"HX-Push-Url": push}, **kwargs)
+
+
+class Reswap(Response):
+    """Class to specify how the response will be swapped."""
+
+    def __init__(
+        self,
+        content: T,
+        method: Literal[
+            "innerHTML", "outerHTML", "beforebegin", "afterbegin", "beforeend", "afterend", "delete", "none"
+        ],
+        **kwargs: Any,
+    ) -> None:
+        """Initialize"""
+        super().__init__(content=content, headers={"HX-Reswap": method}, **kwargs)
+
+
+class Retarget(Response):
+    """Class to target different element on the page"""
+
+    def __init__(self, content: T, target: str, **kwargs: Any) -> None:
+        """Initialize"""
+        super().__init__(content=content, headers={"HX-Retarget": target}, **kwargs)
+
+
+class TriggerEvent(Response):
+    """Trigger Client side event"""
+
+    def __init__(
+        self, content: T, name: str, after: EventAfterType, params: "Dict[str, Any] | None" = None, **kwargs: Any
+    ) -> None:
+        """Initialize"""
+        params = params if params else {}
+        if after == "receive":
+            header = "HX-Trigger"
+        elif after == "settle":
+            header = "HX-Trigger-After-Settle"
+        elif after == "swap":
+            header = "HX-Trigger-After-Swap"
+        else:
+            raise ValueError("Invalid value for after param. Value must be either 'receive', 'settle' or 'swap'.")
+        headers = {header: json.dumps({name: params})}
+        super().__init__(content=content, headers=headers, **kwargs)
+
+
+class HXLocation(Response):
+    """Client side redirect without full page reload."""
+
+    def __init__(
+        self,
+        redirect_to: str,
+        source: "str | None" = None,
+        event: "str | None" = None,
+        target: "str | None" = None,
+        swap: Literal[
+            "innerHTML",
+            "outerHTML",
+            "beforebegin",
+            "afterbegin",
+            "beforeend",
+            "afterend",
+            "delete",
+            "none",
+            None,
+        ] = None,
+        headers: "Dict[str, Any] | None" = None,

```

Comment:
Use python 3.9 compatible `Optional` instead of the pipe operator:

```python
Optional[Dict[str, Any]]
```

Applies everywhere